### PR TITLE
Account the time to deliver connection pool events as part of the reported duration

### PR DESCRIPTION
**Pull request by stIncMale**
The spec change: https://github.com/mongodb/specifications/commit/c6621d01b37df9c7a9ae2036f10d0b6597047bc3.

JAVA-5121

### CODE REVIEW

File: driver-core/src/main/com/mongodb/internal/connection/DefaultConnectionPool.java [hunk 1]
```diff
@@ -504,19 +504,19 @@ private void connectionPoolCreated(final ConnectionPoolListener connectionPoolLi
...
      */
     private TimePoint connectionCreated(final ConnectionPoolListener connectionPoolListener, final ConnectionId connectionId) {
+        TimePoint tConnectionCreated = TimePoint.now();
         logEventMessage("Connection created",
                 "Connection created: address={}:{}, driver-generated ID={}",
                 connectionId.getLocalValue());
 
         connectionPoolListener.connectionCreated(new ConnectionCreatedEvent(connectionId));
-        return TimePoint.now();
+        return tConnectionCreated;
```
Comment (stIncMale) [inline driver-core/src/main/com/mongodb/internal/connection/DefaultConnectionPool.java:519]: This change, and a similar change in the `connectionCheckoutStarted` method are the only behavioral changes in the PR.

File: driver-core/src/main/com/mongodb/internal/connection/DefaultConnectionPool.java [hunk 2]
```diff
@@ -194,34 +194,34 @@ public InternalConnection get(final OperationContext operationContext) {
 
     @Override
     public InternalConnection get(final OperationContext operationContext, final long timeoutValue, final TimeUnit timeUnit) {
-        TimePoint checkoutStart = connectionCheckoutStarted(operationContext);
-        Timeout timeout = Timeout.started(timeoutValue, timeUnit, checkoutStart);
+        TimePoint tConnectionCheckOutStarted = connectionCheckoutStarted(operationContext);
```
Comment (jyemin) [inline driver-core/src/main/com/mongodb/internal/connection/DefaultConnectionPool.java:197]: What's up with the `t` prefix?  Is it Hungarian notation?  It's not something we generally use in this project.

Comment (stIncMale) [reply driver-core/src/main/com/mongodb/internal/connection/DefaultConnectionPool.java:197]: I simply used the name used in pseudocode in the spec, to facilitate matching them.

Comment (jyemin) [reply driver-core/src/main/com/mongodb/internal/connection/DefaultConnectionPool.java:197]: I'd prefer that we stick to our own naming conventions.

Comment (stIncMale) [reply driver-core/src/main/com/mongodb/internal/connection/DefaultConnectionPool.java]: Done.

**Review by jyemin — APPROVED** [review]

**Comment by jyemin** [comment]
One reviewer is sufficient for this.  Removed @rozza

**Pull request merged**